app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryListRecyclerAdapter.kt

```diff
@@ -79,8 +89,12 @@ class ContentEntryListRecyclerAdapter(var itemListener: ContentEntryListItemList
         holder.itemView.setSelectedIfInList(item, selectedItems, ContentEntryList2Fragment.DIFF_CALLBACK)
         if(item != null) {
             GlobalScope.launch(Dispatchers.Main.immediate) {
-                holder.downloadJobItemLiveData = containerDownloadManager
-                        .getDownloadJobItemByContentEntryUid(item.contentEntryUid )
+                holder.downloadJobItemLiveData = RateLimitedLiveData(appDatabase, listOf("ContentJobItem"), 1000) {
+                    appDatabase.contentJobItemDao.findStatusForActiveContentJobItem(item.contentEntryUid)
+                }
+                holder.contentJobItemProgress = RateLimitedLiveData(appDatabase, listOf("ContentJobItem"), 1000) {
```

Comment:
Can't we use an object with embedded fields to do one query instead of two queries? Or would that make things more complicated on the binder or something?